src/applications/vaos/new-appointment/newAppointmentFlow.js

```diff
@@ -58,6 +58,41 @@ function getFacilityPageKey(state) {
   return vaosFlatFacilityPage(state) ? 'vaFacilityV2' : 'vaFacility';
 }
 
+async function vaFacilityNext(state, dispatch) {
+  const eligibilityStatus = getEligibilityStatus(state);
+
+  if (eligibilityStatus.direct) {
+    dispatch(startDirectScheduleFlow());
+    return 'clinicChoice';
+  }
+
+  if (eligibilityStatus.request) {
+    dispatch(startRequestAppointmentFlow());
+    return 'requestDateTime';
+  }
+
+  throw new Error('Veteran not eligible for direct scheduling or requests');
+}
+
+function vaFacilityPrevious(state) {
+  let nextState = 'typeOfCare';
+  const communityCareEnabled = vaosCommunityCare(state);
+
+  if (isSleepCare(state)) {
+    nextState = 'typeOfSleepCare';
+  } else if (
+    communityCareEnabled &&
+    isCCEligible(state) &&
+    getTypeOfCare(getFormData(state))?.ccId !== undefined
+  ) {
+    nextState = 'typeOfFacility';
+  } else if (isEyeCare(state)) {
+    nextState = 'typeOfEyeCare';
+  }
+
+  return nextState;
+}
```

Comment:
Both the existing and new facility pages share the same `next` and `previous` logic, so created a common function for them to share.